### PR TITLE
Converts integrity check dialog to JavaFX

### LINKED ISSUE #2181: Integrity checker should link to entries via reference
Follow up form #1779 

Linking to Bibentries inside the integrity checker should be done via a reference to the entry not the key.
We should switch it to BibEntry, i.e. just keep a reference to the entry and the use/create a method highlight(BibEntry)

### PR DESCRIPTION
**Pull request by tobiasdiez**
Moreover:
- Show entry by reference and not by id. Fixes #2181.
- Fixes a few issues that occurred when opening the entry editor by code from the integrity dialog
- Reuse gridpane in entry editor (should have a slightly superior performance)
- Improve display of progress dialog
- Invoke copy files task using central task executor

image

Filtering (triggered by right-clicking the column):
image

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [x] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I would prefer the search as a field above the table but I'm happy that you added searching! For the checkboxes you probably need a select all / unselect all button to make it easier to select all items

File: src/main/java/org/jabref/gui/actions/CopyFilesAction.java [hunk 1]
```diff
@@ -64,7 +54,12 @@ public void execute() {
             databaseContext = frame.getCurrentBasePanel().getBibDatabaseContext();
 
             Task<List<CopyFilesResultItemViewModel>> exportTask = new CopyFilesTask(databaseContext, entries, path);
-            startServiceAndshowProgessDialog(exportTask);
+            dialogService.showProgressDialogAndWait(
+                    Localization.lang("Copy linked files to folder..."),
+                    Localization.lang("Copy linked files to folder..."),
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/actions/CopyFilesAction.java]: Have you checked that this still runs and does not block?

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/actions/CopyFilesAction.java]: I fixed the canceling of the task now as well.

**Comment by tobiasdiez** [comment]
Thanks for the help @Siedlerchr

**Pull request merged**